app/javascript/dashboard/components-next/message/chips/File.vue

```diff
@@ -16,15 +16,41 @@ const { t } = useI18n();
 
 const fileName = computed(() => {
   const url = attachment.dataUrl;
-  if (url) {
-    const filename = url.substring(url.lastIndexOf('/') + 1);
-    return filename || t('CONVERSATION.UNKNOWN_FILE_TYPE');
+  if (!url) return t('CONVERSATION.UNKNOWN_FILE_TYPE');
+
+  try {
+    const encodedFilename = url.substring(url.lastIndexOf('/') + 1);
+    return decodeURIComponent(encodedFilename);
+  } catch {
+    return t('CONVERSATION.UNKNOWN_FILE_TYPE');
   }
-  return t('CONVERSATION.UNKNOWN_FILE_TYPE');
 });
 
-const fileType = computed(() => {
-  return fileName.value.split('.').pop();
+const fileType = computed(() => fileName.value.split('.').pop()?.toLowerCase());
+
+const fileNameWithoutExt = computed(() => {
```

Comment:
@iamsivin Can we move this to utils?

Reply:
Sure

Reply:
@pranavrajs Here is the PR https://github.com/chatwoot/chatwoot/pull/10819